x-pack/solutions/observability/plugins/observability_logs_explorer/public/redirect_to_discover.tsx

```diff
@@ -25,9 +40,43 @@ export const renderDiscoverRedirect = (core: CoreStart, appParams: AppMountParam
   };
 };
 
-export const DiscoverRedirect = ({ core }: { core: CoreStart }) => {
+export const DiscoverRedirect = ({
+  core,
+  discover,
+}: {
+  core: CoreStart;
+  discover: DiscoverStart;
+}) => {
   const location = useLocation();
-  const path = `${location.pathname}${location.search}`;
-  core.application.navigateToApp(DISCOVER_APP_ID, { replace: true, path });
+  const searchParams = new URLSearchParams(location.search);
+  const pageStateEncoded = searchParams.get('pageState') || '';
+  const rawPageState = safeDecode(pageStateEncoded) || {};
+  const parsedState = normalizeUrlState(rawPageState);
+  if (parsedState) {
+    const dataViewSpec = parsedState.dataSourceSelection
+      ? hydrateDataSourceSelection(parsedState.dataSourceSelection)
+      : hydrateDataSourceSelection(DEFAULT_ALL_SELECTION);
+
+    const discoverParams: DiscoverAppLocatorParams = {
+      indexPatternId: dataViewSpec.id,
```

Comment:
Testing this again I noticed that for the "single" and unresolved selection cases we need to pass the full `dataViewSpec` here.

You could take a look at the [`DiscoverLink`](https://github.com/elastic/kibana/blob/a8a07dfc586d78b8f4b7997b00e126363d68c043/x-pack/packages/observability/logs_overview/src/components/discover_link/discover_link.tsx#L36-L41) again for reference. It pre-dates the "all" and "dataview" cases, though. For these passing the id as you did here is correct AFAIK.